src/test/java/com/theelixrlabs/healthcare/controller/PatchDoctorControllerTest.java

```diff
@@ -0,0 +1,100 @@
+import org.mockito.MockitoAnnotations;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import java.util.UUID;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+/**
+ * Class contains unit tests for the PatchDoctorController in the application.
+ * uses MockMvc to simulate HTTP requests and verify the behavior of the controller methods.
+ * The tests cover scenarios - successful patches and handling invalid UUIDs.
+ */
+@WebMvcTest(controllers = PatchDoctorController.class)
+@AutoConfigureMockMvc(addFilters = false)
+public class PatchDoctorControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private PatchDoctorService patchDoctorService;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    private UUID uuid;
+    private String doctorId;
+    private DoctorDto doctorDto;
+
+    /**
+     * Initializes common test data and mocks before each test.
+     * @throws Exception if setup fails
+     */
+    @BeforeEach
+    void setUp() throws Exception {
+        uuid = UUID.fromString("44fdebfc-7f49-455f-b98b-7d5eef4fe4fb");
+        doctorId = uuid.toString();
+        doctorDto = DoctorDto.builder()
+                .id(uuid)
+                .firstName("Shahana")
+                .lastName("Sha")
+                .department("Psychology")
+                .aadhaarNumber("956788563467")
+                .build();
+        MockitoAnnotations.initMocks(this);
+    }
+
+    /**
+     * Tests a successful PATCH request to update a doctor's details.
+     * Mocks the service to return a DoctorDto object and verifies the response.
+     * @throws Exception if assertion or request fails
+     */
+    @Test
+    public void patchDoctorById_SuccessfulUpdate_ReturnsUpdatedDoctor() throws Exception {
+        when(patchDoctorService.patchDoctorById(doctorId, doctorDto)).thenReturn(doctorDto);
+        ResultActions response = mockMvc.perform(MockMvcRequestBuilders.patch("/doctor/" + doctorId)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(objectMapper.writeValueAsString(doctorDto)));
+        response.andExpect(status().isOk())
+                .andExpect(jsonPath("$.responseData.firstName", CoreMatchers.is(doctorDto.getFirstName())))
```

Comment:
Please create constants

Reply:
instead of mapping each field i used objectmapper to convert java objects into JSON strings for comparison. So constants removed.